app/controllers/news_controller.rb

```diff
@@ -0,0 +1,24 @@
+class NewsController < ApplicationController
+  before_action :set_page
+
+  def show
+    @versions = Version.joins(:rubygem)
+      .recent
+      .indexed
+      .by_created_at
+      .paginate(page: @page, per_page: 10, total_entries: 100)
+  end
+
+  def popular
+    @title = "New Releases — Popular Gems"
+    popular_gem_ids = Rubygem.by_downloads.limit(100).pluck(:id).uniq
```

Comment:
@olivierlacan Can you get away with making this a single sql query?

Sorry if you already tried this. Or if it performs worse.
I tend to lean a little too far towards fewer queries (vs simpler queries).

```ruby
    popular_gems = Rubygem.by_downloads.limit(100) # stays as a scope
    @versions = Version.recent
      .indexed
      .by_created_at
      .where(rubygems: popular_gems)
      .paginate(page: @page, per_page: 10, total_entries: 100)
```